internal/uvm/create.go

```diff
@@ -389,3 +391,42 @@ func (uvm *UtilityVM) CloseGCSConnection() (err error) {
 	}
 	return
 }
+
+// parseStorageSpaceEVDPath verifies the host path provided for the mount
+// type "ExtensibleVirtualDisk" and ExtensibleVirtualDiskType "space"
+// The correct path is in the "evd://space/{pool_guid}{disk_guid}" format.
+func parseStorageSpaceEVDPath(path string) error {
+	if !strings.HasPrefix(path, `evd://space/{`) ||
+		!strings.HasSuffix(path, "}") {
+		return errors.New("storage space evd path must follow 'evd://space/{pool_guid}{disk_guid}' format")
+	}
+	pool_guid_start := len(`evd://space/{`)
+	pool_guid_end := pool_guid_start + 36 // length of GUID string
+	disk_guid_start := pool_guid_end + 2  // skips '}{'
+	disk_guid_end := disk_guid_start + 36
+	pool_guid := path[pool_guid_start:pool_guid_end]
+	disk_guid := path[disk_guid_start:disk_guid_end]
+	if _, err := guid.FromString(pool_guid); err != nil {
+		return fmt.Errorf("pool guid is not valid: %w", err)
+	}
+	if _, err := guid.FromString(disk_guid); err != nil {
+		return fmt.Errorf("disk guid is not valid: %w", err)
+	}
+	return nil
+}
+
+// ParseExtensibleVirtualDiskPath verifies that the `hostPath` provided is a valid extensible virtual
+// disk path. If it is a valid path then returns the `ExtensibleVirtualDiskType` of this mount and the
+// path that should be provided in the mount request to the uvm.
+func ParseExtensibleVirtualDiskPath(hostPath string) (evdType, mountPath string, err error) {
+	if strings.HasPrefix(hostPath, `evd://space/`) {
+		if err = parseStorageSpaceEVDPath(hostPath); err != nil {
+			return
+		}
+		evdType = "space"
+		mountPath = strings.TrimPrefix(hostPath, `evd://space/`)
+	} else {
+		err = fmt.Errorf("unsupported Extensible virtual disk: %s", hostPath)
+	}
+	return
+}
```

Comment:
Would this make more sense to be in scsi.go?

Reply:
Moved.